webauthn4j-core/src/main/java/com/webauthn4j/validator/AuthenticationDataValidator.java

```diff
@@ -233,6 +236,47 @@ private void validate(@NonNull AuthenticationData authenticationData, @NonNull A
         //spec| If all the above steps are successful, continue with the authentication ceremony as appropriate. Otherwise, fail the authentication ceremony.
     }
 
+    void validateClientPaymentData(CollectedClientAdditionalPaymentData clientAdditionalPaymentData, PaymentAuthenticationParameters paymentAuthenticationParameters) {
+        AssertUtil.notNull(clientAdditionalPaymentData, "clientAdditionalPaymentData must not be null during payment");
+        AssertUtil.notNull(paymentAuthenticationParameters, "paymentAuthenticationParameters must not be null during payment");
+
+        PaymentServerProperty paymentServerProperty = (PaymentServerProperty) paymentAuthenticationParameters.getServerProperty();
+        AssertUtil.notNull(paymentServerProperty, "paymentServerProperty must not be null during payment");
+
+        // validate the rpId
+        if (!Objects.equals(clientAdditionalPaymentData.getRp(), paymentAuthenticationParameters.getServerProperty().getRpId())) {
+            throw new BadRpIdException("payment rpId does not match server value");
+        }
+
+        // validate top origin
+        originValidator.validate(new Origin(clientAdditionalPaymentData.getTopOrigin()), paymentServerProperty.getOrigins());
+
+        // validate payee origin
+        originValidator.validate(new Origin(clientAdditionalPaymentData.getPayeeOrigin()), paymentServerProperty.getPayeeOrigins());
```

Comment:
Since some library users may want to validate topOrigin and payeeOrigin with custom logic rather than comparing with origin list, it is better to delegates to a new dedicated interface (`PaymentOriginValidator` ?)